integration-tests/tests/cargo_near_new.rs

```diff
@@ -0,0 +1,180 @@
+        &tests_manifest(),
+        &generated_manifest,
+        "near_workspaces",
+    )?;
+
+    let opts = cargo_near_build::docker::DockerBuildOpts::builder()
+        .manifest_path(generated_manifest.clone())
+        .context(cargo_near_build::BuildContext::Build)
+        .build();
+
+    let artifact = cargo_near_build::docker::build(opts)?;
+
+    let contract_wasm = std::fs::read(artifact.path)?;
+
+    test_basics_on(&contract_wasm).await?;
+
+    std::fs::remove_dir_all(
+        generated_manifest
+            .parent()
+            .expect("expected to have parent"),
+    )?;
+
+    Ok(())
+}
+
+// this item ensures that tests of this module share logic of `test_basics_on`
+// with `../../cargo-near/src/commands/new/new-project-template/tests/test_basics.rs`
+include! {"../../cargo-near/src/commands/new/test_basics_on.rs.in"}
+
+/// This runs `cargo test` on project, generated by `cargo near new`
+#[test]
+fn test_cargo_test_on_generated_project() -> Result<(), Box<dyn std::error::Error>> {
+    use cargo_near_integration_tests::setup_tracing;
+
+    setup_tracing();
+
+    let generated_manifest = run_cargo_near_new()?;
+
+    toolchain_channels::assert_equal(
+        &tests_toolchain(),
+        &generated_manifest
+            .parent()
+            .expect("has a parent")
+            .join("rust-toolchain.toml"),
+    )?;
+
+    // somehow `rustup` doesn't respect `rust-toolchain.toml` in
+    // the target generated dir
+    let mut cmd = Command::new("cargo");
+    cmd.arg("test")
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .current_dir(
+            generated_manifest
+                .parent()
+                .expect("expected to have parent")
+                .as_str(),
+        );
+
+    let (output, output_string) = {
+        let child = cmd.spawn()?;
+        let output = child.wait_with_output()?;
+        let mut output_string = String::new();
+        output_string.push_str(&String::from_utf8_lossy(&output.stderr));
+        output_string.push_str(&String::from_utf8_lossy(&output.stdout));
+        (output, output_string)
+    };
+
+    assert!(
+        output.status.success(),
+        "running `cargo test` errored with msg: {}",
+        output_string
+    );
+
+    std::fs::remove_dir_all(
+        generated_manifest
+            .parent()
+            .expect("expected to have parent"),
+    )?;
+
+    Ok(())
+}
+
+/// tests logic of `../../cargo-near/src/commands/new/new-project-template/tests/test_basics.rs`,
+/// on contract, which is built by [cargo_near_build::build] from project, generated by `cargo near new`
+///
+/// differs from [test_cargo_test_on_generated_project] in the aspect, that current test
+/// uses [cargo_near_build::build] logic from current *cargo-near* branch, whereas
+/// [test_cargo_test_on_generated_project] uses [cargo_near_build::build] logic from
+/// latest published [near_workspaces::compile_project]  
```

Comment:
this test addresses suggestion in https://github.com/near/cargo-near/pull/229#issuecomment-2399227236